src/cljs/main/broadfcui/common/table/table.cljs

```diff
@@ -0,0 +1,141 @@
+                                                       :rows results
+                                                       :loading? false))}))
+   :get-initial-state
+   (fn [{:keys [props]}]
+     (assoc
+      (persistence/try-restore
+       {:key (:state-key props)
+        :validator (fn [stored-value]
+                     (or (not (:v props))
+                         (= (:v props) (:v stored-value))))
+        :initial
+        #(let [columns (-> props :body :columns)
+               initial-sort-column (or (first (filter :sort-initial columns))
+                                       (when-not (-> props :body :behavior :allow-no-sort?)
+                                         (first columns)))
+               initial-sort-order (when initial-sort-column
+                                    (get initial-sort-column :sort-initial :asc))]
+           (merge
+            {:query-params (select-keys
+                            {:page-number 1
+                             :rows-per-page 20
+                             :filter-text ""
+                             :sort-column (table-utils/resolve-id initial-sort-column)
+                             :sort-order initial-sort-order}
+                            (difference all-query-params (-> props :body :external-query-params)))
+             :column-display (table-utils/build-column-display columns)}
+            (when-let [v (:v props)] {:v v})))})
+       :rows []))
+   :render
+   (fn [{:keys [props state]}]
+     (let [props (utils/deep-merge default-props props)
+           {:keys [rows column-display total-count filtered-count query-params]} @state
+           {:keys [body toolbar paginator]} props
+           {:keys [empty-message columns behavior external-query-params]} body
+           {:keys [fixed-column-count allow-no-sort?]} behavior
+           query-params (merge query-params (select-keys props external-query-params))
+           update-column-display #(swap! state assoc :column-display %)]
+       [:div {}
+        (when (:loading? @state)
+          [comps/DelayedBlocker {:banner "Loading..."}])
+        [:div {:style (:style toolbar)}
+         (when (:reorderable-columns? behavior)
+           (let [button-props (:column-edit-button toolbar)]
+             [:div {:style (:style button-props)}
+              [ColumnEditButton
+               (assoc (utils/restructure columns column-display update-column-display fixed-column-count)
+                 :reorder-anchor (:anchor button-props)
+                 :button (:button button-props))]]))
+         (when (and (:filterable? behavior) (not (contains? external-query-params :filter-text)))
+           (let [filter-bar-props (:filter-bar toolbar)]
+             [:div {:style (:style filter-bar-props)}
+              [comps/TextFilter {:on-filter #(swap! state update :query-params assoc :filter-text %)}]]))
+         (list* (:items toolbar))]
```

Comment:
So the user can specify the additional toolbar items as a vector.  `list*` converts to a list so that rendering doesn't try to treat the toolbar items array as a single DOM element.